widget/list.go

```diff
@@ -412,8 +423,12 @@ func (l *listLayout) Layout(objects []fyne.CanvasObject, size fyne.Size) {
 }
 
 func (l *listLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
-	return fyne.NewSize(l.list.itemMin.Width,
-		l.list.itemMin.Height*l.list.Length())
+	size := fyne.NewSize(0, 0)
+	if f := l.list.Length; f != nil {
+		size = fyne.NewSize(l.list.itemMin.Width,
+			l.list.itemMin.Height*f())
+	}
+	return size
```

Comment:
Using named return values saves you a line:
```go
func (l *listLayout) MinSize(objects []fyne.CanvasObject) (size fyne.Size) {
	if f := l.list.Length; f != nil {
		size.Width = l.list.itemMin.Width
		size.Height = l.list.itemMin.Height*f()
	}
	return
}
```

Reply:
Personally I would just invert the orignal logic.
```
if l.list.Length == nil {
    return fyne.Size{}
}

return itemMin etc...
```

That's exiting early and putting the logic at the top level.

Reply:
but not guarding against the Length variable changing which is the point of the whole f :=

Reply:
True - are we supporting the functions changing? I guess it makes sense.

You could remove a line of code by not having the size varable and just replacing the assign and return with two returns.

Reply:
resolved